src/app/store/country.reducer.ts

```diff
@@ -35,5 +37,17 @@ export const countryReducer = createReducer(
     ...state,
     countries: [...countries],
     error: null,
+  })),
+  // Search countries by region
+  on(searchCountriesByRegionSuccess, (state, { countries }) => ({
+    ...state,
+    countries: [...countries],
+    error: null,
+  })),
+  // Search countries by code
+  on(searchCountriesByCodeSuccess, (state, { countries }) => ({
```

Comment:
In addition to error scenario we need to handle the pending state here and show loader here.